test/integration/master_test_utils.go

```diff
@@ -0,0 +1,198 @@
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/master"
+	"github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/admission/admit"
+	"github.com/golang/glog"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"time"
+)
+
+// Timeout used in benchmarks, to eg: resize an rc
+var Timeout = 30 * time.Minute
+
+// TODO: Include all master components (scheduler, nodecontroller).
+// TODO: Reconcile with integration.go, currently the master used there doesn't understand
+// how to restart cleanly, which is required for each iteration of a benchmark. The integration
+// tests also don't make it easy to isolate and turn off components at will.
+type MasterComponents struct {
+	// Raw http server in front of the master
+	apiServer *httptest.Server
+	// Kubernetes master, contains an embedded etcd helper
+	kubeMaster *master.Master
+	// Restclient used to talk to the kubernetes master
+	restClient *client.Client
+	// Replication controller manager
+	controllerManager *controller.ReplicationManager
+	// Channel for stop signals to all master components
+	stopCh chan struct{}
+}
+
+// TODO: Run these in containers each with their own --parent-cgroup so they don't step on each other?
+func NewMasterComponents() *MasterComponents {
+	m, s := startMasterOrDie()
+	glog.Infof("Master %+v", s.URL)
+	deleteAllEtcdKeys()
+	restClient := client.NewOrDie(&client.Config{Host: s.URL, Version: "v1beta3"})
+	stopCh := make(chan struct{})
+	controllerManager := controller.NewReplicationManager(restClient)
+	controllerManager.RunUntil(1*time.Second, stopCh)
+	return &MasterComponents{
+		apiServer:         s,
+		kubeMaster:        m,
+		restClient:        restClient,
+		controllerManager: controllerManager,
+		stopCh:            stopCh,
+	}
+}
+
+// startMasterOrDie starts a kubernetes master and an httpserver to handle api requests
+func startMasterOrDie() (*master.Master, *httptest.Server) {
+	helper, err := master.NewEtcdHelper(newEtcdClient(), "")
+	if err != nil {
+		panic(err)
+	}
+	var m *master.Master
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		m.Handler.ServeHTTP(w, req)
+	}))
+
+	m = master.New(&master.Config{
+		EtcdHelper:        helper,
+		KubeletClient:     client.FakeKubeletClient{},
+		EnableLogsSupport: false,
+		EnableProfiling:   true,
+		EnableUISupport:   false,
+		APIPrefix:         "/api",
+		Authorizer:        apiserver.NewAlwaysAllowAuthorizer(),
+		AdmissionControl:  admit.NewAlwaysAdmit(),
+		SkipServices:      true,
+	})
+	return m, s
+}
+
+func (m *MasterComponents) Stop() {
+	glog.Infof("Stopping master components")
+	// Ordering matters because the apiServer will only shutdown when pending
+	// requests are done
+	close(m.stopCh)
+	m.apiServer.Close()
+}
+
+// StopRc stops the rc via kubectl's stop library
+func (m *MasterComponents) StopRc(rc *api.ReplicationController) error {
```

Comment:
StartRC/StopRC here feel like they should be standalone functions that take a client.

(should be RC not Rc, btw)